src/main/org/deidentifier/arx/aggregates/HierarchyBuilderDate.java

```diff
@@ -534,10 +534,11 @@ private String generalize(String input, Granularity granularity) {
         if (_range == null) {
             return sdf.format(date);
         } else {
-            int year = Integer.valueOf(sdf.format(date));
-            int lower = (year / _range) * _range;
-            int upper = ((year / _range) + 1) * _range;
-            return "[" + lower + ", " + upper + "[";
+            int dateUnit = Integer.valueOf(sdf.format(date));
+            int lower    = Integer.valueOf((dateUnit - 1) / (_range))  * (_range) + 1;
+            int upper    = lower + _range - 1;
+            String outputDate = "[" + lower + "-" + upper + "]";

```

Comment:
The return format [1-3] breaks with the old format which would be [1, 4[. Please adjust. Line 540 can remain unchanged. line 539 should be rewritten as upper    = lower + _range;